File: cozumler/tekrar-eden-harfleri-bulmaca/javascript/tekrar-eden-harfleri-bulmaca.js
const multipleOccurencesOfLetters = sentence => {
  const letters = {}
  sentence = sentence.toLowerCase()

  for (let i = 0; i < sentence.length; i++) {
    const char = sentence[i]

    if (letters.char || !char.match(/[a-z]/)) continue

    letters[char] = sentence.match(new RegExp(char, 'g')).length
  }
  
  return Object.keys(letters)
    .filter(k => letters[k] > 1)
    .map(k => ({ [k]: letters[k] }))
}

console.log(multipleOccurencesOfLetters('Tekrar eden harfleri bulmaca!'))
// [ { e: 4 }, { r: 4 }, { a: 4 }, { l: 2 } ]

File: cozumler/tekrar-eden-harfleri-bulmaca/ocaml/tekrar-eden-harfleri-bulmaca.ml
module SMap = Map.Make(String)

(* Tüm karakterleri sayar (boşluk ve noktalamalar dahil) . Büyük-Küçük harf ayrımı yapmaz. *)
let rec countChars str =
  if String.length str > 0
  then SMap.merge
    ( fun key a b -> match a, b with
    | (x, None  ) -> x
    | (None  , b) -> b
    | (Some x, Some y) -> Some (x + y)
    )
    ( countChars (String.sub str 1 (String.length str - 1)) )
    ( SMap.add (String.sub str 0 1) 1 SMap.empty )
  else SMap.empty

(* 1 adet olanları filtreler. *)
let countRepeatedChars str =
  SMap.filter (fun key value -> value > 1) (countChars str)

let () =
  let test = "Tekrar eden harfler sorusu icin test dizgesi." in
  SMap.iter
    (fun key count -> Printf.printf "%s : %d\n" key count)
    (countRepeatedChars test)
